src/inmanta/ast/type.py

```diff
@@ -704,6 +968,23 @@ def get_namespace(self) -> "Namespace":
     def get_double_defined_exception(self, other: "NamedType") -> DuplicateException:
         return DuplicateException(self, other, "TypeConstraint %s is already defined" % (self.get_full_name()))
 
+    def get_base_type(self) -> "Type":
+        assert self.basetype is not None
+        return self.basetype
```

Comment:
Also doesn't seem to completely match the docstring (though it's somewhat ambiguous here). I notice it's used in the lsm module `__init__.py` and `typedef.py`. Have you checked that those are compatible?

Reply:
it is indeed wrong.